Neos.ContentRepository.NodeAccess/Classes/FlowQueryOperations/RemoveOperation.php

```diff
@@ -0,0 +1,83 @@
+<?php
+namespace Neos\ContentRepository\NodeAccess\FlowQueryOperations;
+
+/*
+ * This file is part of the Neos.ContentRepository package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\ContentRepository\Core\Projection\ContentGraph\Filter\FindReferencesFilter;
+use Neos\ContentRepository\Core\Projection\ContentGraph\Node;
+use Neos\ContentRepositoryRegistry\ContentRepositoryRegistry;
+use Neos\Eel\FlowQuery\FlowQuery;
+use Neos\Eel\FlowQuery\OperationInterface;
+use Neos\Flow\Annotations as Flow;
+
+/**
+ * Removes the given Node from the current context.
+ *
+ * The operation accepts one argument that may be an Array, a FlowQuery
+ * or an Object.
+ *
+ * The result is an array of {@see Node} instances.
+ *
+ * @api To be used in Fusion, in php this should be implemented directly
+ */
+final class RemoveOperation implements OperationInterface
```

Comment:
i dislike the `remove` naming. I think we should stick more to vocabulary of other functional libraries.

Maybe a filter? `nodes.filter((i) => !myNode.equals(i))`

Reply:
There is remove in flowQuery already see `https://github.com/neos/flow-development-collection/blob/9.0/Neos.Eel/Classes/FlowQuery/Operations/RemoveOperation.php`. Only problem is that it does not work reliably on Nodes anymore. That is why i added a nodeSpecific implementation.

Reply:
Okay sorry I forgot… then it’s okay I guess, and even needed for compatibility?

I’d like this info in the doc block somewhere that we override this op.

Reply:
Added a hint in the docblock as requested